python/foglamp/plugins/south/modbustcp/modbustcp.py

```diff
@@ -175,9 +175,13 @@ def plugin_poll(handle):
                 source_port = int(handle['port']['value'])
             except:
                 raise ValueError
-
-            mbus_client = ModbusTcpClient(host=source_address, port=source_port)
-            _LOGGER.info('Modbus TCP Client is connected: %s, %s:%d', mbus_client.connect(), source_address, source_port)
+            try:
+                mbus_client = ModbusTcpClient(host=source_address, port=source_port)
+            except:
+                _LOGGER.warn('Failed to connect! Modbus TCP host %s on port %d', source_address, source_port)
+                return

```

Comment:
Shall we not raise an exception, e.g RunTimeError, with a proper error message?

Reply:
here it should return withOUT any action; as connection has failed, we logged it. and it will continue to check with poll routine